sklearn_pandas/pipeline.py

```diff
@@ -3,12 +3,37 @@
 from sklearn.utils import tosequence
 
 
+def _call_fit(fit_method, X, y=None, **kwargs):
+    """
+    helper function, calls the fit or fit_transform method with the correct
+    number of parameters
+
+    fit_method: fit or fit_transform method of the transformer
+    X: the data to fit
+    y: the target vector relative to X, optional
+    kwargs: any keyword arguments to the fit method
+
+    return: the result of the fit or fit_transform method
+
+    WARNING: if this function raises a TypeError exception, test the fit
+    or fit_transform method passed to it in isolation as _call_fit will not
+    distinguish TypeError due to incorrect number of arguments from
+    other TypeError
+    """
+    try:
+        return fit_method(X, y, **kwargs)
+    except TypeError:
```

Comment:
What about:
```

except TypeError as e:
    if e.args and 'takes exactly' in e.args[0]:
        # fit takes only one argument
        return fit_method(X, **kwargs)
    # TypeError caused by some other unkown reason
    raise
```

I know it looks a bit hacky but I guess it will solve your warning above.